### ISSUE TITLE
"Comparison to null is expensive" warning can be misleading

### ISSUE THREAD

**Issue by cathei** [JetBrains/resharper-unity#2365]
The warning, [Avoid null comparisons against UnityEngine.Object subclasses](https://github.com/JetBrains/resharper-unity/wiki/Avoid-null-comparisons-against-UnityEngine.Object-subclasses) shows for this code:
```
if (obj != null) DoSomething();
```
But not showing for this code:
```
if (obj) DoSomething();
```
Since bool cast operator of `UnityEngine.Object` does the same thing as comparing to `null`, the warning should be shown as same. Otherwise, it can mislead users to use bool cast instead, which is worse practice as it is less explicit.

**Comment by NikolayTimofeev** [comment]
Yeah, they both use the same CompareBaseObjects() inside, no difference in performance.
The warning is misleading for sure.

**Comment by citizenmatt** [comment]
Do you mean the plain underline highlight for performance related info inside `Update` and related methods?

**Comment by NikolayTimofeev** [comment]
Yes
